components/kyma-environment-broker/internal/storage/dbsession/read.go

```diff
@@ -43,6 +44,26 @@ func (r readSession) GetOperationByID(opID string) (OperationDTO, dberr.Error) {
 	return operation, nil
 }
 
+func (r readSession) GetOperationsInProgressByType(operationType OperationType) ([]OperationDTO, dberr.Error) {
+	stateCondition := dbr.Eq("state", domain.InProgress)
+	typeCondition := dbr.Eq("type", operationType)
+	var operations []OperationDTO
+
+	_, err := r.session.
+		Select("*").
+		From(postsql.OperationTableName).
+		Where(stateCondition).
+		Where(typeCondition).
+		Load(&operations)
+	if err != nil {
+		if err == dbr.ErrNotFound {
```

Comment:
if there is no any in progress operations, it is better to return empty list. Such case is not an error.